### PR TITLE
Fix 49: Create standalone pinny

### PR DESCRIPTION
**Pull request by hcstephencheung**
Attempt to fix https://github.com/mobify/pinny/issues/49

Added custom shim configuration scripts for jQuery and Velocity so that pinny.js can be a standalone script.

### CODE REVIEW

**Comment by scalvert** [comment]
Hey @hcstephencheung! Thanks for the PR. Can you give us a bit more detail on the problem this is trying to solve? As it stands, both Velocity and jQuery are already AMD compatible, so I don't think we need those wrappers. Perhaps I misunderstand what this PR is for.

**Comment by hcstephencheung** [comment]
@scalvert There was an issue trying to compile pinny with jQuery/Velocity excluded in the build. The custom shims were aimed to fix those compilation errors. @tedtate wanted to create a standalone pinny so that users can install pinny with their own versions of jQuery/Velocity.

**Comment by tedtate** [comment]
@scalvert This is trying to make it easier for Non-AMD users to use Pinny. Right now you would need to include something like ~8 scripts to make pinny work. This PR moves that number down to 2 (jQuery and Velocity).

File: src/js/jquery_wrapper.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+if ( typeof define === 'function' && define.amd && define.amd.jQuery ) {
+	define('$', (function (global) {
+		return function () {
+			var ret, fn;
+			return ret || global.$;
```
Comment (ericmuyser) [inline src/js/jquery_wrapper.js]: This should probably be `global.jQuery`, to be in line with the `window.jQuery` below (support for noConflict).

Actually, why not `return ret || global.jQuery || global.Zepto`, removing the `&& define.amd.jQuery` check, and rename this file to `selector_library_wrapper.js`?

Comment (tedtate) [inline src/js/jquery_wrapper.js]: +1 on all of @ericmuyser 's suggestions.

**Comment by scalvert** [comment]
So help me understand. You're saying r.js needs this to build a standalone file? Also, jQuery and Velocity are both AMD compatible, so why do we need these wrappers?

**Comment by donnielrt** [comment]
@tedtate could we not use require.js' `map` for this?

http://requirejs.org/docs/jquery.html#noconflictmap

**Comment by scalvert** [comment]
@tedtate @hcstephencheung OK thanks for explaining. It makes sense. 

@donnielrt may be onto something there though - using map may work, as from my understanding it's for mapping names to a different alias.

**Comment by tedtate** [comment]
I think that is going to run into the same issue. You aren't going to be able to use `map` because we aren't actually including jQuery inside of our build.

Do you have time to verify if `map` will help us out @hcstephencheung?

**Comment by hcstephencheung** [comment]
@donnielrt it seems that even with map, we'll need to map the jquery to our own defined $ module (which is what the wrapper module is doing). Also, Zepto isn't AMD compatible so we'll still need to properly define $ to use whatever is available in global (assuming users are still free to use jQuery or Zepto)

@tedtate I can't really think of another solution to achieve the 2 problems you stated above, so perhaps this is the best solution to go with?

@scalvert other thoughts?

**Comment by donnielrt** [comment]
Hey, we’re looking to prune older unattended PRs. If this PR is still relevant and you would like to see it merged in, please reopen the PR and we’ll add it to our backlog! Thanks!

**Pull request closed without merging**